### PR TITLE
Asynchronous loadflow calculation

### PR DESCRIPTION
**Pull request by anistouri**

### CODE REVIEW

File: src/components/run-button-container.js [hunk 1]
```diff
@@ -180,27 +181,36 @@ export function RunButtonContainer({
...
-                setComputationStopped(!computationStopped);
-            } else if (action === runnable.DYNAMIC_SIMULATION) {
-                setDynamicSimulationStatusState(RunningStatus.IDLE);
-                stopDynamicSimulation(studyUuid, currentNode?.id);
-                setComputationStopped(!computationStopped);
-            } else if (action === runnable.VOLTAGE_INIT) {
-                setVoltageInitStatusState(RunningStatus.IDLE);
-                stopVoltageInit(studyUuid, currentNode?.id);
-                setComputationStopped(!computationStopped);
+            switch (action) {
```
Comment (flomillot) [inline src/components/run-button-container.js:184]: :+1:

**Comment by etiennehomer** [comment]
The flows are not updated on the geographical image after a loadflow.
Even after a refresh of the page, we don't have it. Maybe it's on the back side that something is missing.

File: src/components/utils/running-status.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ export function getLoadFlowRunningStatus(
     loadFlowStatus: string
 ): RunningStatus {
     switch (loadFlowStatus) {
-        case 'CONVERGED':
+        case 'COMPLETED':
```
Comment (etiennehomer) [inline src/components/utils/running-status.ts:19]: I think we want to keep CONVERGED and DIVERGED. It's an important notion for the loadflow

File: src/components/diagrams/diagram-pane.js [hunk 3]
```diff
@@ -1098,7 +1097,6 @@ export function DiagramPane({
                                     diagramView.svgType ===
                                         DiagramType.SUBSTATION) && (
                                     <SingleLineDiagramContent
```
Comment (etiennehomer) [inline src/components/diagrams/diagram-pane.js:1099]: Here you need the loadflow status to set the flows in disable or not

**Review by etiennehomer — DISMISSED** [review]

File: src/components/utils/running-status.ts (hunk 2 above)
Comment (anistouri) [reply src/components/utils/running-status.ts]: done

**Review by flomillot — APPROVED** [review]
Code OK

File: src/components/loadflow-result-tab.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+export const LoadFlowResultTab = ({ studyUuid, nodeUuid }) => {
+    const [loadflowResult, isWaiting] = useNodeData(
+        studyUuid,
+        nodeUuid,
+        fetchLoadFlowResult,
+        loadflowResultInvalidations
+    );
+
+    const [loadflowStatus] = useNodeData(
```
Comment (flomillot) [inline src/components/loadflow-result-tab.js:28]: Why still use this and not useSelector ?

**Review by flomillot — CHANGES_REQUESTED** [review]

**Review by SlimaneAmar — APPROVED** [review]

File: src/components/loadflow-result-tab.js (hunk 4 above)
Comment (anistouri) [reply src/components/loadflow-result-tab.js]: Done

File: src/components/diagrams/diagram-pane.js (hunk 3 above)
Comment (anistouri) [reply src/components/diagrams/diagram-pane.js]: Done

**Pull request merged**